client/my-sites/plans/main.jsx

```diff
@@ -55,25 +53,21 @@ var Plans = React.createClass( {
 	},
 
 	comparePlansLink: function() {
-		var url = '/plans/compare',
-			selectedSite = this.props.sites.getSelectedSite();
-
-		var compareString = this.translate( 'Compare Plans' );
+		var selectedSite = this.props.site,
+			compareString;
 
 		if ( selectedSite.jetpack ) {
 			compareString = this.translate( 'Compare Options' );
+		} else {
+			compareString = this.translate( 'Compare Plans' );
 		}
 
 		if ( this.props.plans.get().length <= 0 ) {
 			return '';
 		}
 
-		if ( selectedSite ) {
-			url += '/' + selectedSite.slug;
-		}
-
 		return (
-			<a href={ url } className="compare-plans-link" onClick={ this.recordComparePlansClick }>
+			<a href={ '/plans/compare/' + selectedSite.slug } className="compare-plans-link" onClick={ this.recordComparePlansClick }>
```

Comment:
This line works because `selectedSite` is never `undefined`, and `( false ).name === undefined`, right? If that is the case, we might want to change it, as this seems somewhat fragile.